build-llvm.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019 The ClangBuiltLinux Authors
+# Description: Builds an LLVM toolchain suitable for kernel development
+
+import argparse
+import datetime
+import pathlib
+import os
+import subprocess
+import shutil
+import textwrap
+import time
+import utils
+
+
+# Returns Clang's version as an integer
+#
+# Some backstory:
+#     When this was written in bash, this was implemented along the lines of
```

Comment:
If this part is simpler is bash, feel free to move just this part back to shell script, and have Python invoke it, if you find that to be simpler and more concise.  The point of the rewrite is to have a minimal amount of shell script to maintain, but always choose the best tool for the job.

Reply:
I considered it but I'd rather not import a shell script when this works fine on every distribution that I tried.

Reply:
https://github.com/MSF-Jarvis/tc-build/commit/658107da11ed4c7ac3503ca8dada587c274c55bf